imageio/imageio-jpeg/src/main/java/com/twelvemonkeys/imageio/plugins/jpeg/JPEGImageReader.java

```diff
@@ -711,13 +861,34 @@ private void initHeader() throws IOException {
 
             this.segments = segments;
 
+            readExifOrientation();
+
             if (DEBUG) {
                 System.out.println("segments: " + segments);
                 System.out.println("Read metadata in " + (System.currentTimeMillis() - start) + " ms");
             }
         }
     }
 
+    private void readExifOrientation() throws IOException {
+        EXIF exif = getExif();
+        if (exif != null) {
+            CompoundDirectory dict = parseExif(exif);
+            
+            if (dict != null) {
+                Iterator<Entry> iterator = dict.iterator();
+                while(iterator.hasNext()) {
+                    Entry entry = iterator.next();
+                    if (entry.getIdentifier().equals(TIFF.TAG_ORIENTATION)) {
+                        exifOrientation = ((Integer)entry.getValue()).intValue();
```

Comment:
```suggestion
                        exifOrientation = ((Number) entry.getValue()).intValue();
```

is a bit safer. The tag is defined to contain a `SHORT`, but could also be `LONG`. 

(and to be honest, the low-level TIFF stuff is a bit weird at the moment, it should be fixed/refactored so that we could just call `entry.getIntValue()`, my bad...).

Reply:
OK, will adjust this as well.